install/0000_90_machine-api-operator_04_alertrules.yaml

```diff
@@ -40,3 +28,13 @@ spec:
             severity: critical
           annotations:
             message: "machine api MAO metric {{ $labels.kind }} is not being reported"
+    - name: mao-down
```

Comment:
Personally I think we should expand M-A-O. It's easier for newcomers/users/customers/et al to sound it out, then guess what it stands for.

Reply:
I agree, but personally won't block this PR on it. We should try to merge this today.

My main question is: Doesn't this new alert make the existing `MachineMAOMetricsDown` alert unnecessary? If so, should we remove it?

Reply:
No, it does not make MachineMAOMetricsDown unnecessary. This alert is for overall mao process. While the other one is specific to metrics reporting. It may happen that MAO is not down but metrics are not being reported for any reason.

Reply:
Ah, so the `mapi_mao_collector_up` metric tracks whether we're successfully scraping operands, or MachineSets etc.?

Reply:
yes or any other similar problem which is specific to metrics reporting.

Reply:
If you have problems inside a team identifying what `mapi_mao_collector_up` refers to, how is customer supposed to know it?

Also a good practice is to expose metrics by operand itself and not aggregate them inside an operator. For example look how openshift or kube apiserver does it.

Reply:
@paulfantom regarding naming confusion, have added more details and improved the naming.
MAO has also been updated to "machine-api-operator" at all the instances

Regarding reporting from operand, we discussed this part internally and decided to report cluster state related metrics from the operator. operands will also be reporting metrics specific to operands in the future.